pkg/build/api/v1/types.go

```diff
@@ -402,6 +402,11 @@ type SourceBuildStrategy struct {
 
 	// ForcePull describes if the builder should pull the images from registry prior to building.
 	ForcePull bool `json:"forcePull,omitempty"`
+
+	// disableImplicitBuild prevents source-to-image from performing a Docker build operation
```

Comment:
Why do you need to do it on a per user basis?  Adding something to the API
is burdensome, it doesn't match cleanly to actual security requirements,
and it's just adding the wrong sorts of complexity.  If we had proper
network support then this would not be an issue.

I'm not in favor of opening it up as an API field.

On Thu, Apr 28, 2016 at 2:21 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/api/v1/types.go
> <https://github.com/openshift/origin/pull/8664#discussion_r61477153>:
>
> > @@ -402,6 +402,11 @@ type SourceBuildStrategy struct {
> >
> >  	// ForcePull describes if the builder should pull the images from registry prior to building.
> >  	ForcePull bool `json:"forcePull,omitempty"`
> > +
> > +	// disableImplicitBuild prevents source-to-image from performing a Docker build operation
>
> @smarterclayton <https://github.com/smarterclayton> fair enough, but I
> want to be able to control this behavior on a per user basis so we need
> some way to convey to s2i what is/isn't allowed. I'm open to suggestions,
> but I started by putting it on the build pod directly until I realized the
> build pods are managed by the controller which means they can't check
> permissions in a useful way.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8664/files/4251157e4e6c6d4b865d530bb0de37eb6fd75b3a#r61477153>
>